cmd/transcriber/call/utils.go

```diff
@@ -0,0 +1,139 @@
+package call
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+const (
+	httpRequestTimeout         = 5 * time.Second
+	httpUploadTimeout          = 10 * time.Second
+	maxUploadRetryAttempts     = 5
+	uploadRetryAttemptWaitTime = 5 * time.Second
+)
+
+func (t *Transcriber) getUserForSession(sessionID string) (*model.User, error) {
+	ctx, cancelFn := context.WithTimeout(context.Background(), httpRequestTimeout)
+	defer cancelFn()
+
+	url := fmt.Sprintf("%s/plugins/%s/bot/calls/%s/sessions/%s/profile", t.cfg.SiteURL, pluginID, t.cfg.CallID, sessionID)
+	resp, err := t.apiClient.DoAPIRequest(ctx, http.MethodGet, url, "", "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch user profile: %w", err)
+	}
+	defer resp.Body.Close()
+
+	var user *model.User
+	if err := json.NewDecoder(resp.Body).Decode(&user); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal user profile: %w", err)
+	}
+
+	return user, nil
+}
+
+func getDataDir() string {
+	if dir := os.Getenv("DATA_DIR"); dir != "" {
+		return dir
+	}
+	return dataDir
+}
+
+func getModelsDir() string {
+	if dir := os.Getenv("MODELS_DIR"); dir != "" {
+		return dir
+	}
+	return modelsDir
+}
+
+func (t *Transcriber) publishTranscription(f *os.File) (err error) {
+	if _, err := f.Seek(0, 0); err != nil {
+		return fmt.Errorf("failed to seek: %w", err)
+	}
+
+	apiURL := fmt.Sprintf("%s/plugins/%s/bot", t.apiClient.URL, pluginID)
+
+	info, err := f.Stat()
+	if err != nil {
+		return fmt.Errorf("failed to stat file: %w", err)
+	}
+
+	us := &model.UploadSession{
+		ChannelId: t.cfg.CallID,
+		Filename:  filepath.Base(f.Name()),
+		FileSize:  info.Size(),
+	}
+
+	payload, err := json.Marshal(us)
+	if err != nil {
+		return fmt.Errorf("failed to encode payload: %w", err)
+	}
+
+	for i := 0; i < maxUploadRetryAttempts; i++ {
```

Comment:
Hmm, I've never had to do this before, but what happens if a req fails halfway through? When you retry are the re-requests (for ones that were successful the first time around) going to cause problems?

Reply:
Ah, right, new session each time. 🤦